File: sport/src/main/java/david/microservices/sport/Logic.java
package david.microservices.sport;

import org.springframework.beans.factory.annotation.Autowired;

import org.springframework.stereotype.Component;


import java.util.List;

@Component
public class Logic {

    @Autowired
    private SportDataJpaRespository sportDataJpaRespository;

    public void deleteSportsData(long id) { sportDataJpaRespository.deleteById(id); }

    public void insert(SportData sportData) {
        sportDataJpaRespository.save(sportData);
    }

    public List<SportData> getAll() {
        return sportDataJpaRespository.findAll();
    }

    public List<SportData> getUserSportsById(int id) {
        return sportDataJpaRespository.findByUserID(id);
    }

    public List<SportData> getSportsType(int id, String type) {
        return sportDataJpaRespository.findByUserIDAndType(id, type);
    }
}

File: zielsetzungen/src/main/kotlin/david/schweizer/zielsetzungen/Logic.kt
package david.schweizer.zielsetzungen

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class Logic {
    @Autowired
    lateinit var objectiveJpaRespository: ObjectiveJpaRespository

    fun deleteObjective(id: Long) = objectiveJpaRespository.deleteById(id)

    fun getAllObjectives(): MutableList<Objective> = objectiveJpaRespository.findAll()

    fun getObjectivesByUserIdAndType(userId: Int, type: String) = objectiveJpaRespository.findByUserIDAndType(userId, type)

    fun getObjectivesByUserId(userId: Int) = objectiveJpaRespository.findByUserID(userId)

    fun insertData(data: Objective) = objectiveJpaRespository.save(data)
}